tools/prow-config-updater/updater.go

```diff
@@ -0,0 +1,189 @@
+				// Best effort, won't fail the process if the comment fails.
+				cu.githubcommenter.commentOnStagingFailure(*pr.Number, err)
+				return fmt.Errorf("error running Prow staging process: %v", err)
+			} else {
+				// Best effort, won't fail the process if the comment fails.
+				cu.githubcommenter.commentOnStagingSuccess(*pr.Number, err)
+			}
+			// TODO(chizhg): create an auto-merge pull request to update production Prow configs.
+		} else {
+			if err := cu.updateProw(config.ProdProwEnv); err != nil {
+				return fmt.Errorf("error updating production Prow configs: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+// Update Prow with the changed config files and send message for the update result.
+func (cu *ConfigUpdater) updateProw(env config.ProwEnv) error {
+	updatedFiles, err := cu.doProwUpdate(env)
+	prnumber := cu.prnumber
+	if err == nil {
+		// Best effort, won't fail the process if the comment fails.
+		cu.githubcommenter.commentOnUpdateConfigsSuccess(prnumber, config.ProdProwEnv, updatedFiles)
+	} else {
+		// Best effort, won't fail the process if the comment fails.
+		cu.githubcommenter.commentOnUpdateConfigsFailure(prnumber, config.StagingProwEnv, updatedFiles, err)
+	}
+	return err
+}
+
+// Decide if we need staging process by checking the changed files.
+func (cu *ConfigUpdater) needsStaging() bool {
+	// If any key config files for staging Prow are changed, the staging process will be needed.
+	fs := collectRelevantFiles(cu.files, config.StagingProwKeyConfigPaths)
+	return len(fs) != 0
+}
+
+// Start running the staging process to update and test staging Prow.
+func (cu *ConfigUpdater) startStaging() error {
+	// Update staging Prow.
+	if err := cu.updateProw(config.StagingProwEnv); err != nil {
+		return fmt.Errorf("error updating staging Prow configs: %v", err)
+	}
+	// Create pull request in the fork repository for the testing of staging Prow.
+	fpr, err := cu.githubmainhandler.createForkPullRequest()
+	if err != nil {
+		return fmt.Errorf("error creating pull request in the fork repository: %v", err)
+	}
+	// Create comment on the fork pull request to get it tested by staging Prow and merged.
+	forkprnumber := *fpr.Number
+	if err = cu.githubcommenter.commentToMergePullRequest(forkprnumber); err != nil {
+		return fmt.Errorf("error creating comment on the fork pull request: %v", err)
+	}
+	// Wait for the fork pull request to be automatically merged by staging Prow.
+	if err := cu.githubmainhandler.waitForForkPullRequestMerged(*fpr.Number); err != nil {
+		return fmt.Errorf("error waiting for the fork pull request to be merged: %v", err)
+	}
+	return nil
+}
+
+// Filter out all files that are under the given paths.
+func collectRelevantFiles(files []string, paths []string) []string {
+	rfs := make([]string, 0)
+	for _, f := range files {
+		for _, p := range paths {
+			if strings.HasPrefix(f, p) {
+				rfs = append(rfs, f)
+			}
+		}
+	}
+	return rfs
+}
+
+// Run the `make` command to update Prow configs.
```

Comment:
Did you see this PR? https://github.com/knative/test-infra/pull/1815

Reply:
Yes I saw it, I have added a comment to replace the command once that PR got merged.